doc/development/guide/tests.rst

```diff
@@ -16,26 +15,156 @@ These requirements can be installed via ``pip``:
 
     pip install pytest pytest-cov pytest-mock flaky
 
+Creating a test
+~~~~~~~~~~~~~~~
+Every test has to be added to the PennyLane `test folder <https://github.com/PennyLaneAI/pennylane/tree/master/tests>`__.
+The test folder is following PennyLane structure, and therefore a test needs to be added to its corresponding folder.
+
+Most tests do not require any interface (Autograd, Torch, TensorFlow and Jax). Tests without interface will be marked
+as ``core`` tests automatically by pytest ``conftest.py``. For general tests, you can follow this example structure,
+where it is only required to follow pytest guidelines:
+
+.. code-block:: python
+
+    import pennylane as qml
+
+    def test_circuit_expval(self):
+        """ Test that the circuit expectation value for PauliX is 0."""
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.expval(qml.PauliX(0))
+
+        expval = circuit()
+
+        assert expval == 0
+
+This test will be marked automatically as a ``core`` test.
+
+On the other hand, some tests require specific interfaces and need to marked in order to be run on our Github test suite.
+Tests involving interfaces have to be marked with its respective marker: ``@pytest.mark.autograd``, ``@pytest.mark.torch``,
+``@pytest.mark.tf`` and ``@pytest.mark.jax``. If tests involved multiple interfaces, one should add the marker
+``@pytest.mark.all_interfaces``. It is now prohibited to use ``importorskip`` inside tests with interfaces. Also tests
+involving interfaces must be written separately and it should be avoided to use fixtures. The necessary packages related
+to interfaces have to be imported in the tests directly. All tests with marked interfaces are skipped if the necessary
+interfaces are not installed.
+
+Tests that are not marked and also related to specific interfaces will be skipped in the Github test suite and therefore
+it will not be run and the results will not be added to the coverage (inducing a coverage fail).
```

Comment:
Re our discussion, I think the missing marker here would lead to a failed test run, since the import statement should raise an error.
```suggestion
Tests that are not marked but do import an interface will lead to a failure in the Github test suite.
```